spec/mailboxes/imap/imap_mailbox_spec.rb

```diff
@@ -264,5 +264,54 @@
         expect(conversation.additional_attributes['in_reply_to']).to eq(multiple_in_reply_to_mail.in_reply_to.first)
       end
     end
+
+    context 'when a reply to a conversation started by an agent' do
+      let(:agent_conversation) { create(:conversation, account: account, inbox: channel.inbox, assignee: agent) }
+      let(:reply_mail_with_fallback_reference) do
+        # Simulate an email reply with a reference that matches FALLBACK_PATTERN
+        reference_id = "account/#{account.id}/conversation/#{agent_conversation.uuid}@chatwoot.com"
+        create_inbound_email_from_mail(
+          from: 'email@gmail.com',
+          to: 'imap@gmail.com',
+          subject: 'Re: Agent started conversation',
+          references: [reference_id]
+        )
+      end
+
+      it 'appends email to the existing conversation using FALLBACK_PATTERN' do
+        expect(agent_conversation.messages.size).to eq(0)
+
+        class_instance.process(reply_mail_with_fallback_reference.mail, channel)
```

Comment:
I'm not sure if this is a good approach.

Reply:
For the reference matching you mean?